game-core/src/main/java/games/strategy/triplea/ui/history/HistoryPanel.java

```diff
@@ -447,12 +450,13 @@ public Component getTreeCellRendererComponent(
       if (value instanceof Step) {
         final PlayerId player = ((Step) value).getPlayerId();
         if (player != null) {
+          final String text = value.toString() + " (" + player.getName() + ")";
           if (uiContext != null) {
             super.getTreeCellRendererComponent(tree, value, sel, expanded, leaf, row, haveFocus);
             icon.setImage(uiContext.getFlagImageFactory().getSmallFlag(player));
             setIcon(icon);
+            setToolTipText(text);
           } else {
```

Comment:
Side note, it seems odd for UI text to be null, I wonder if that is a dead code branch :thinking: